codecov.yml

```diff
@@ -23,15 +23,64 @@ coverage:
       default:
         branches:
           - develop
+        target: 90%
         informational: true
         only_pulls: true
         paths:
-          - "nncf/onnx"
-          - "nncf/common"  # extend this once we collect coverage reports for more than just onnx and common part of precommit
+          - "nncf"
 
 comment:
-  layout: "diff, flags, files"
+  layout: "reach, diff, files, flags, components"
```

Comment:
Now the comment is overloaded with useless information and can be fully views only on vertical monitor.

![image](https://github.com/openvinotoolkit/nncf/assets/48012821/11672cd6-1860-4511-a074-e29c6ee7b559)

Looks like only `diff, files` are usefully. May be `components` can be interesting too.

Reply:
"Useless" is subjective, and you have not yet had a chance to determine which of the new sections are actually useful or not. Memorize your objections for when the time comes to disable the sections that we will consider "useless" based on experience, perhaps via voting.

Reply:
If you think that it can be useful, provide scenarios how it can help in development.

Reply:
The component section gives a quick rundown on which parts of NNCF the change impacts the most.

Of the flags section, only the COMMON is important really, and it shows the part of `common` code that is tested by `common` tests in precommit - we should encourage PRs that raise this number up to the target level of 80-90%. Sadly, it is not possible to only show a specific flag in the comment, so it shows everything. This may be circumvented by the `flag_regexes` [option](https://docs.codecov.com/docs/components#component-options), e.g. a `common_tested_by_common_only` component may be configured to only limit itself to the `COMMON` flag and hopefully show the same numbers as the current `COMMON` flag coverage, but this is an improvement for the future.

The diff section is the one that shows the coverage percentage of the whole project. If the comment text itself showed this already, maybe we wouldn't need it.

The files section gives a per-file breakdown of the coverage impact and the links to the corresponding per-line coverage visualization.

I do have doubts about how useful the reach graph really is, but we haven't seen it in action yet on real PRs, so we cannot jump to conclusions just now. Presumably it should help visually identify the PRs that preserve global and component coverage, but significantly reduce coverage in one file and increase the coverage in another. Maybe there also features hidden in this graph that will be helpful that we cannot yet see (since no feature overview is provided by codecov).